etc/run_with_override.jl

```diff
@@ -12,21 +12,23 @@ gapoverride = abspath(gapoverride)
 #
 @info "Install needed packages"
 using Pkg
-Pkg.add(["GAP_jll"])
-Pkg.add(["GAP_lib_jll"])
 Pkg.develop(path=dirname(dirname(@__FILE__)))
+Pkg.add(["GAP_jll", "GAP_lib_jll"])
 Pkg.instantiate()
 
+import GAP_lib_jll
+
 #
 #
 #
-function add_jll_override(depot, pkgname, newdir)
+function add_jll_override(depot, pkgname, pkguuid, newdir)
     pkgid = Base.identify_package("$(pkgname)_jll")
-    uuid = string(pkgid.uuid)
+    # does not work with julia 1.12-dev
+    # pkguuid = string(pkgid.uuid)
```

Comment:
This threw some (seemingly random) segfaults with nightly. But only if running from the console, not if copying everything to the reply.
Thus I took the easy solution and added the UUID to the arguments instead

Reply:
Just tried with Julia master and when running this code I actually got
```
ERROR: LoadError: type Nothing has no field uuid
```
which would suggest `pkgid === nothing`, but then also the `touch` below would be borked?

Reply:
if you print the pkgid, this succeeds (and prints something non-trivial), but i then get a segfault instead of a LoadError when accessing pkgid.uuid

Reply:
Indeed: printing the `pkgid` makes it proceed -- that seems like a big regression in Julia, printing a value shouldn't change it form `nothing` to something, should it? In particular since *this* works in the REPL (without printing):
```julia
pkgid = Base.identify_package("GAP_jll"); pkgid.uuid
```
Perhaps we should open an issue?

Anyway, so I inserted the `print` and it proceeds, but I don't get a crash, I get this:
```
[ Info: Created temporary depot at /var/folders/d_/1zss2fnd6xdgclqnj8jj_27m0000gp/T/jl_7ABU3q
pkgname GAP has pkgid Base.PkgId(Base.UUID("5cd7a574-2c56-5be2-91dc-c8bc375b9ddf"), "GAP_jll")
pkgname GAP_lib has pkgid Base.PkgId(Base.UUID("de1ad85e-c930-5cd4-919d-ccd3fcafd1a3"), "GAP_lib_jll")
Precompiling GAP...
Info Given GAP was explicitly requested, output will be shown live
ERROR: LoadError: ArgumentError: Package GAP_jll [5cd7a574-2c56-5be2-91dc-c8bc375b9ddf] is required but does not seem to be installed:
 - Run `Pkg.instantiate()` to install all recorded dependencies.
```
which makes no sense, as we run `Pkg.instantiate()`. But I've inserted another `Pkg.instantiate()` invocation right after the `add_jll_override` calls.

Reply:
Thanks for [reporting it upstream](https://github.com/JuliaLang/julia/issues/54599), it seems we've hit an IMHO rather serious looking code generation bug in Julia.

For now we can keep this workaround, but let's remove it again when they fix their bug.